flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -2607,6 +2612,38 @@ mlir::Value IntrinsicLibrary::genSpacing(mlir::Type resultType,
       Fortran::lower::genSpacing(builder, loc, fir::getBase(args[0])));
 }
 
+// SPREAD
+fir::ExtendedValue
+IntrinsicLibrary::genSpread(mlir::Type resultType,
+                            llvm::ArrayRef<fir::ExtendedValue> args) {
+
+  assert(args.size() == 3);
+
+  // Handle source argument
+  auto source = builder.createBox(loc, args[0]);
+  fir::BoxValue sourceTmp = source;
+  auto sourceRank = sourceTmp.rank();
+  assert(sourceRank < 15);
```

Comment:
Similar comment as to Pete's recent PR ... shouldn't the rank be checked in the front end? I wouldn't expect anything to reach lowering with a rank that was invalid.

One advantage of trusting semantics to get this right on its own is that if someone comes along and wants to add a switch to support ranks up to _n_ then lowering can just be agnostic to the issue (for the most part).